services/usb-livesync-service-base.ts

```diff
@@ -78,70 +78,48 @@ export class UsbLiveSyncServiceBase implements IUsbLiveSyncServiceBase {
-			let platformLowerCase = platform.toLowerCase();
-			let synciOSSimulator = this.$hostInfo.isDarwin && platformLowerCase === "ios" && (this.$options.emulator || this.$iOSEmulatorServices.isSimulatorRunning().wait());
+			let synciOSSimulator = this.shouldSynciOSSimulator(data.platform).wait();
 
 			if(synciOSSimulator) {
-				this.$iOSEmulatorServices.sync(appIdentifier, projectFilesPath, notRunningiOSSimulatorAction).wait();
+				this.$iOSEmulatorServices.sync(data.appIdentifier, data.projectFilesPath, data.notRunningiOSSimulatorAction).wait();
 			}
 
-			if(!this._initialized && (!this.$options.emulator || platform.toLowerCase() === "android")) {
-				this.initialize(platform).wait();
+			if(!this._initialized && (!this.$options.emulator || data.platform.toLowerCase() === "android")) {
+				this.initialize(data.platform).wait();
 			}
 
-			if(!this.$options.emulator || platform.toLowerCase() === "android") {
-				let projectFiles = this.$fs.enumerateFilesInDirectorySync(projectFilesPath,
-					(filePath, stat) => !this.isFileExcluded(path.relative(projectFilesPath, filePath), excludedProjectDirsAndFiles, projectFilesPath),
+			if(!this.$options.emulator || data.platform.toLowerCase() === "android") {
+				let projectFiles = this.$fs.enumerateFilesInDirectorySync(data.projectFilesPath,
+					(filePath, stat) => !this.isFileExcluded(path.relative(data.projectFilesPath, filePath), data.excludedProjectDirsAndFiles, data.projectFilesPath),
 					{ enumerateDirectories: true }
 				);
 
-				this.syncCore(platform,
-					projectFiles,
-					appIdentifier,
-					localProjectRootPath || projectFilesPath,
-					platformSpecificLiveSyncServices,
-					notInstalledAppOnDeviceAction,
-					beforeLiveSyncAction
-					).wait();
+				this.syncCore(data, projectFiles, false).wait();
 			}
 
 			if(this.$options.watch) {
+
 				let that = this;
 				this.$hooksService.executeBeforeHooks('watch').wait();
-				gaze("**/*", { cwd: watchGlob }, function(err: any, watcher: any) {
+
+				gaze("**/*", { cwd: data.watchGlob }, function(err: any, watcher: any) {
 					this.on('all', (event: string, filePath: string) => {
 						if(event === "added" || event === "changed") {
-							if(!that.isFileExcluded(filePath, excludedProjectDirsAndFiles, projectFilesPath)) {
-								if(synciOSSimulator) {
-									that.batchSimulatorLiveSync(
-										appIdentifier,
-										projectFilesPath,
-										filePath,
-										notRunningiOSSimulatorAction,
-										iOSSimulatorRelativeToProjectBasePathAction
-									);
+							if(!that.isFileExcluded(filePath, data.excludedProjectDirsAndFiles, data.projectFilesPath)) {
+								let canExecuteFastLiveSync = data.canExecuteFastLiveSync && data.canExecuteFastLiveSync(filePath);
+
+								if(synciOSSimulator && !canExecuteFastLiveSync) {
+									that.batchSimulatorLiveSync(data, filePath);
+								}
+
+								if( (!that.$options.emulator || data.platform.toLowerCase() === "android") && !canExecuteFastLiveSync) {
```

Comment:
`(!that.$options.emulator || data.platform.toLowerCase() === "android")` is used many times, can we extract it to a variable